client/src/components/common/boost/BoostCalibration.tsx

```diff
@@ -0,0 +1,86 @@
+import React , {useCallback, useState} from 'react';
+import {Button, Card, InputGroup, FormControl} from 'react-bootstrap';
+import {setCalibration as sendCalibrationValue, resetCalibration} from 'api/common/powerModel';
+// import {useSensorData} from 'api/v2/sensors/data';
+
+/**
+ * Boost distance claibration input form
+ * 
+ * @returns {React.Component} Component
+ */
+function BoostCalibrationInput() {
+    const [calibrationValue, setCalib] = useState('');
+
+    const handleCalibrationChange = useCallback(
+        (event) => {
+        setCalib(event.target.value);
+        },
+        [setCalib],
+    );
+
+    const onSet = useCallback(
+        () => {
+          const calibValue: number = parseInt(calibrationValue, 10);
+          if (Number.isNaN(calibValue)) {
+            alert("Please enter an integer value only!");
+          }
+          else {
+            console.log(calibValue);
+            sendCalibrationValue(parseInt(calibrationValue, 10));
+          }
+          setCalib('');
+        },
+        [calibrationValue, setCalib],
+    );
+
+    const handleKeyPressed = useCallback(
+        (event) => { if (event.key === 'Enter') onSet(); },
+        [onSet]
+    );
+
+    return (
+    <InputGroup className="mb-1" >
+        <FormControl
+            onChange={handleCalibrationChange}
+            onKeyPress={handleKeyPressed}
+            placeholder="Calibrate distance..."
+            value={calibrationValue}
+        />
+        <InputGroup.Append className="l5">
+            <Button className="mr-2" variant="outline-secondary" onClick={onSet}>Set</Button>
+        </InputGroup.Append>
+        
+        <Button variant="outline-danger" onClick={resetCalibration}>Reset</Button>
+    </InputGroup>
+    );
+}
+
+/**
+ * Boost calibration card, displays the calibrated distance and the 
+ * distance travelled by the bike
+ * 
+ * @returns {React.Component} Component
+ */
+export default function BoostCalibration() {
+    // const GetDistTravelled = () => {
+    //     return useSensorData("reed_distance").data;
+    // };
```

Comment:
Remove unused code - this will be passed in via a prop anyway

Reply:
Should it passed down or I was thinking maybe `BoostCalibration` can just read it off? If it gets passed down, it would have to be passed down by `BoostView` right?

Reply:
Correct, you'll need to pass it down. The data will be fetched and provided by either `BoostView` or some intermediate container component (I can't remember which we ended up going with elsewhere).